resources/bundles/org.eclipse.core.resources/src/org/eclipse/core/internal/dtree/DeltaDataTree.java

```diff
@@ -951,8 +963,17 @@ public void storeStrings(StringPool set) {
 	/** for debugging purposes only */
 	@Override
 	public String toString() {
-		return this.getClass().getSimpleName() + " " + Integer.toHexString(System.identityHashCode(this)) + " parent=" //$NON-NLS-1$ //$NON-NLS-2$
-				+ (parent == null ? null : Integer.toHexString(System.identityHashCode(parent))) + " rootNode=" //$NON-NLS-1$
-				+ rootNode.toShortString();
+		DeltaDataTree root = getParent();
```

Comment:
Could you please provide before / after state? Why is this needed, what is better now?

Reply:
somewhere in BuildManager.needsBuild() i wanted information about the depth- which according to the ElementTree javadoc may be the root error to my problems - if it grows. But where to write the depth? ElementTree  or DeltaDataTree? as i found that that tree is actually stored in the DeltaDataTree i think the information belongs there.
before:
![image](https://user-images.githubusercontent.com/51790620/188479054-08e741bb-233e-44b2-aa38-307150ef7d57.png)

after:
![image](https://user-images.githubusercontent.com/51790620/188478757-dc87d439-cf8c-4fec-b216-07755d7ddcfc.png)